src/Compilers/CSharp/Test/Semantic/Semantics/OverloadResolutionPerfTests.cs

```diff
@@ -786,5 +786,231 @@ public void DefiniteAssignment_ManySwitchCasesAndLabels()
                 //         int tmp2; // unused
                 Diagnostic(ErrorCode.WRN_UnreferencedVar, "tmp2").WithArguments("tmp2").WithLocation(9, 13));
         }
+
+        [Fact]
+        public void NestedLambdasAndOverloads_NoTypes_NoErrors()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F0(string s0, string s1, string s2) { }
+    static void F1(string s)
+    {
+        F(s, s0 => {
+        F(s, s1 => {
+        F(s, s2 => {
+            F0(s0, s1, s2);
+        });
+        });
+        });
+    }
+    static void F(string s, Action<string> a) { }
+    static void F(string s, Action<string, string> a) { }
+    static void F(string s, Func<string, string> f) { }
+    static void F(object o, Action<object> a) { }
+    static void F(object o, Action<object, object> a) { }
+    static void F(object o, Func<object, object> f) { }
+}";
+            var comp = CreateCompilation(source);
+            var data = new LambdaBindingData();
+            comp.TestOnlyCompilationData = data;
+            var diagnostics = comp.GetDiagnostics();
+            Assert.Equal((84, 126), (data.LambdaBindingCount, data.UnboundLambdaStateCount));
```

Comment:
>(84, 126) [](http://example.com/codeflow?start=25&length=9)

Without the checks in `BuildArgumentsForErrorRecovery`, this success case reports (136, 186).